buildSrc/src/main/kotlin/org/gradle/plugins/reporting/CiReportingPlugin.kt

```diff
@@ -0,0 +1,91 @@
+package org.gradle.plugins.reporting
+
+import me.champeau.gradle.japicmp.JapicmpTask
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.reporting.Reporting
+import org.gradle.kotlin.dsl.withGroovyBuilder
+import org.gradle.plugin.devel.tasks.ValidateTaskProperties
+import org.gradle.plugins.classycle.Classycle
+import org.gradle.testing.DistributedPerformanceTest
+import org.gradle.testing.DistributionTest
+import java.io.File
+
+/**
+ * When run from a Continuous Integration environment, we only want to archive a subset of reports, mostly for
+ * failing tasks only, to not use up unnecessary disk space on Team City. This also improves the performance of
+ * artifact publishing by reducing the artifacts and packaging reports that consist of multiple files.
+ *
+ * Reducing the number of reports also makes it easier to find the important ones when analysing a failed build in
+ * Team City.
+ */
+open class CiReportingPlugin : Plugin<Project> {
+
+    override fun apply(project: Project) {
```

Comment:
:x: `... apply(project: Project) = project.run {` to avoid one level of nesting.

Reply:
Done.